import/controls/permissionmodel.cpp

```diff
@@ -43,15 +43,22 @@ void PermissionModel::componentComplete()
 
 void PermissionModel::add(const QString &host, const QString &type, int capability)
 {
+    int it = 0;
+    for (const auto &perm : m_permissionList) {
+        if (perm.m_host == host) {
```

Comment:
Should this be checking the `type` too? Otherwise it's impossible to add different permissions for different types on the same host.

Reply:
Is a straight string equality enough to test for matching hosts here? (e.g. is it certain by this point that the transport, port, resource name etc. have been stripped?).